daemon/restart.go

```diff
@@ -35,11 +36,25 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		defer daemon.Unmount(container)
 	}
 
-	if err := daemon.containerStop(container, seconds); err != nil {
+	// set AutoRemove flag to false before stop so the container won't be
+	// removed during restart process
+	autoRemove := container.HostConfig.AutoRemove
```

Comment:
Never mind, there're too many discussions in this page and it's hard to find  necessary informations now :smile: 
Thank you for help review :wink: